src/components/strategies/overlapping/OverlappingBudgetDistribution.tsx

```diff
@@ -0,0 +1,156 @@
+import { Token } from 'libs/tokens';
+import { FC, useId } from 'react';
+import { cn, tokenAmount } from 'utils/helpers';
+import { ReactComponent as IconDeposit } from 'assets/icons/deposit.svg';
+import { ReactComponent as IconWithdraw } from 'assets/icons/withdraw.svg';
+import { WarningMessageWithIcon } from 'components/common/WarningMessageWithIcon';
+import styles from './OverlappingBudgetDistribution.module.css';
+
+interface Props {
+  buy?: boolean;
+  token: Token;
+  initialBudget: string;
+  withdraw: string;
+  deposit: string;
+  balance: string;
+}
+
+function getBudgetDistribution(
+  initial: number,
+  withdraw: number,
+  deposit: number,
+  balance: number
+) {
+  const total = initial + balance;
+  const delta = ((deposit || withdraw) / total) * 100;
+  const newAllocation = (Math.max(initial - withdraw, 0) / total) * 100;
+  const newBalance = (Math.max(balance - deposit, 0) / total) * 100;
+  return {
+    mode: deposit ? 'deposit' : 'withdraw',
+    allocationPercent: Math.round(newAllocation),
+    deltaPercent: Math.round(delta),
+    balancePercent: Math.round(newBalance),
+  };
+}
+
+export const OverlappingBudgetDistribution: FC<Props> = (props) => {
+  const allocatedId = useId();
+  const walletId = useId();
+  const { buy, token, initialBudget, withdraw, deposit, balance } = props;
+  const dist = getBudgetDistribution(
+    Number(initialBudget),
+    Number(withdraw),
+    Number(deposit),
+    Number(balance)
+  );
+  const color = buy ? 'bg-buy' : 'bg-sell';
+  return (
+    <div className="flex flex-col gap-4">
+      <h4 className="text-14 font-weight-500">{buy ? 'Buy' : 'Sell'}</h4>
+      <div className="flex justify-between text-12 text-white/60">
+        <label htmlFor={allocatedId}>
+          Allocated:&nbsp;
+          <span className="text-white">
+            {tokenAmount(initialBudget, token)}
+          </span>
+        </label>
+        <label htmlFor={walletId}>
+          Wallet:&nbsp;
+          <span className="text-white">{tokenAmount(balance, token)}</span>
+        </label>
+      </div>
+      <div className={styles.progress}>
```

Comment:
why not use tailwind classes?

Reply:
In an early version I was using some complex CSS selector and keyframes animations. We simplified everything to use only flex-grow so I think I can move it back to Tailwind indeed👍